lib/WorseReflection/Core/Type/BooleanLiteralType.php

```diff
@@ -27,4 +30,41 @@ public function generalize(): Type
     {
         return new BooleanType();
     }
+
+    public function or(BooleanType $right): BooleanType
+    {
+        if ($right instanceof BooleanLiteralType) {
+            return new self($this->value || $right->value);
+        }
+
+        return new BooleanType();
+    }
+
+    public function and(BooleanType $right): BooleanType
+    {
+        if ($right instanceof BooleanLiteralType) {
+            return new self($this->value && $right->value);
+        }
+
+        return new BooleanType();
+    }
+
+    public function negate(): BooleanType
+    {
+        return new self(!$this->value);
+    }
+
+    public function xor(BooleanType $booleanType): BooleanType
+    {
+        if ($booleanType instanceof BooleanLiteralType) {
+            return new self($this->value() xor $booleanType->value());
+        }
+
+        return new BooleanType();
+    }
+
+    public function toTrinary(): Trinary
```

Comment:
bit of a smell, can probably use the Boolean and BooleanLiteral types instead of `Trianry`